paddle/fluid/framework/new_executor/interpretercore.cc

```diff
@@ -182,6 +183,16 @@ void InterpreterCore::Convert(
     vec_instruction_.emplace_back(op_idx, std::move(op_func_node), *dev_ctx_);
     auto& instr = vec_instruction_.back();
 
+#if defined(PADDLE_WITH_CUDA) || defined(PADDLE_WITH_HIP)
```

Comment:
We can consider to abstract these logic to make it easier to extand more hardware platforms in the furture but not in this PR.

Because if we switch in a new device, we must add `if defined XXX` everywhere.

Reply:
Deeply agree. We need something like "DeviceStream", which unifies the stream of different devices.